ethereum/chain.py

```diff
@@ -345,7 +345,7 @@ def add_transaction(self, transaction):
         log.debug('add tx', num_txs=self.num_transactions(), tx=transaction, on=head_candidate)
         if self.head_candidate.includes_transaction(transaction.hash):
             log.debug('known tx')
-            return
+            return False
```

Comment:
sure - good point 
I'll revert that adjustment and just keep the change here https://github.com/ethereum/pyethereum/pull/341/files#diff-cee411c8659f68f68e4673368f5eae06R366